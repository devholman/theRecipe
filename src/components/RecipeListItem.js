import React from 'react';
import propTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Recipe = ({
  recipe,
  onclick,
  onFavorites,
  favorited,
  isFav,
  ...props
}) => {
  const { name, category, id } = recipe;
  const fav = favorited ? '😀' : '🗃️';

  return isFav ? (
    <Link to={`/recipe/${id}`}>
      <li className="border-bottom border-bottom-dashed py2 pointer">
        <span
          className="mr1"
          role="img"
          aria-label="favorite"
          onClick={e => {
            e.stopPropagation();
            onFavorites(id);
          }}
        >
          {fav}
        </span>
        <span>{name}</span>
        <span>{category}</span>
      </li>
    </Link>
  ) : (
    <li
      className="border-bottom border-bottom-dashed py2 pointer"
      onClick={e => {
        e.stopPropagation();
        onclick(id);
      }}
    >
      <span
        className="mr1"
        role="img"
        aria-label="favorite"
        onClick={e => {
          e.stopPropagation();
          onFavorites(id);
        }}
      >
        {fav}
      </span>
      <span>{name}</span>
      <span>{category}</span>
    </li>
  );
};
Recipe.propTypes = {
  recipes: propTypes.array,
  style: propTypes.object,
  onclick: propTypes.func
};

Recipe.defaultProps = {
  recipe: {},
  isFav: false
};

export default Recipe;
