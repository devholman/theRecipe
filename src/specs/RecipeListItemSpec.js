import React from 'react';
//executes a full render on the component passed by args
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import RecipeListItem from '../components/RecipeListItem';

const testRecipe = {
  id: 1,
  name: 'Chocolate Cake',
  category: 'dessert'
};
describe('<RecipeListItem />', () => {
  test('should not break if no recipe is sent', () => {
    const component = renderer.create(<RecipeListItem />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
  test('should render successfully', () => {
    const component = renderer.create(<RecipeListItem recipe={testRecipe} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
  test('should render favorite state true', () => {
    const component = renderer.create(
      <RecipeListItem recipe={testRecipe} favorited={true} />
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
  test('should render favorite state false', () => {
    const component = renderer.create(
      <RecipeListItem recipe={testRecipe} favorited={false} />
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('should call onClick when clicked', () => {
    const onClick = jest.fn();
    const component = mount(
      <RecipeListItem recipe={testRecipe} onclick={onClick} />
    );
    component.simulate('click');
    expect(onClick.mock.calls.length).toBe(1);
  });
  test('should call Favorited when clicked', () => {
    const onFavorites = jest.fn();
    const component = mount(
      <RecipeListItem recipe={testRecipe} onFavorites={onFavorites} />
    );
    component
      .find('span')
      .first()
      .simulate('click');
    expect(onFavorites.mock.calls.length).toBe(1);
  });
  test('should not call onClick when Favorited is called', () => {
    const onClick = jest.fn();
    const onFavorites = jest.fn();
    const component = mount(
      <RecipeListItem
        recipe={testRecipe}
        onclick={onClick}
        onFavorites={onFavorites}
      />
    );
    component
      .find('span')
      .first()
      .simulate('click');
    expect(onClick.mock.calls.length).toBe(0);
    expect(onFavorites.mock.calls.length).toBe(1);
  });
});
