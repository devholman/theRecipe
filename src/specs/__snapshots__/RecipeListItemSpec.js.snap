// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeListItem /> should not break if no recipe is sent 1`] = `
<li
  className="border-bottom border-bottom-dashed py2 pointer"
  onClick={[Function]}
>
  <span
    aria-label="favorite"
    className="mr1"
    onClick={[Function]}
    role="img"
  >
    🗃️
  </span>
  <span />
  <span />
</li>
`;

exports[`<RecipeListItem /> should render favorite state false 1`] = `
<li
  className="border-bottom border-bottom-dashed py2 pointer"
  onClick={[Function]}
>
  <span
    aria-label="favorite"
    className="mr1"
    onClick={[Function]}
    role="img"
  >
    🗃️
  </span>
  <span>
    Chocolate Cake
  </span>
  <span>
    dessert
  </span>
</li>
`;

exports[`<RecipeListItem /> should render favorite state true 1`] = `
<li
  className="border-bottom border-bottom-dashed py2 pointer"
  onClick={[Function]}
>
  <span
    aria-label="favorite"
    className="mr1"
    onClick={[Function]}
    role="img"
  >
    😀
  </span>
  <span>
    Chocolate Cake
  </span>
  <span>
    dessert
  </span>
</li>
`;

exports[`<RecipeListItem /> should render successfully 1`] = `
<li
  className="border-bottom border-bottom-dashed py2 pointer"
  onClick={[Function]}
>
  <span
    aria-label="favorite"
    className="mr1"
    onClick={[Function]}
    role="img"
  >
    🗃️
  </span>
  <span>
    Chocolate Cake
  </span>
  <span>
    dessert
  </span>
</li>
`;
